src/pymor/vectorarrays/interface.py

```diff
@@ -970,6 +970,16 @@ def from_numpy(self, data, ensure_copy=False):
         """
         raise NotImplementedError
 
+    @property
+    def dim(self):
+        return self._dim
+
+    @dim.setter
+    def dim(self, dim):
+        if not isinstance(dim, int):
```

Comment:
This fails on NumPy integers:

```python
>>> import numpy as np
>>> a = np.int32(5)
>>> a
5
>>> isinstance(a, int)
False
```

Using `numbers.Integral` makes it work:

```python
>>> from numbers import Integral
>>> isinstance(a, Integral)
True
```

Still, that doesn't work with 0-dimensional NumPy arrays, where checking the dimension and dtype seems to be necessary:

```python
>>> b = np.array(5)
>>> b
array(5)
>>> print(b)
5
>>> isinstance(b, Integral)
False
>>> b.ndim == 0 and np.issubdtype(b.dtype, np.integer)
True
```

I'm not sure how far we need to go. Maybe it's ok to just limit to the `Integral` check and see if issues come up.